qiskit/optimization/converters/linear_equality_to_penalty.py

```diff
@@ -16,29 +16,38 @@
 
 import copy
 from typing import Optional, cast, Union, Tuple
+from math import fsum
+import logging
 
 from ..problems.quadratic_program import QuadraticProgram
 from ..problems.variable import Variable
 from ..problems.constraint import Constraint
 from ..problems.quadratic_objective import QuadraticObjective
 from ..exceptions import QiskitOptimizationError
 
+logger = logging.getLogger(__name__)
+
 
 class LinearEqualityToPenalty:
     """Convert a problem with only equality constraints to unconstrained with penalty terms."""
 
-    def __init__(self):
+    def __init__(self, penalty: Optional[float] = None, name: Optional[str] = None):
+        """
+        Args:
+            penalty: Penalty factor to scale equality constraints that are added to objective.
+                     If None is passed, penalty factor will be automatically calculated.
+            name: The name of the converted problem.
+        """
         self._src = None
```

Comment:
To fix mypy It should be:
self._src = None  # type: Optional[QuadraticProgram]
self._dst = None  # type: Optional[QuadraticProgram]